AlphaWallet/Tokens/Types/TokensDataStore.swift

```diff
@@ -642,42 +646,12 @@ class TokensDataStore {
     }
 
     func updatePrices() {
-        guard let priceToUpdate = getPriceToUpdate() else { return }
-        guard !isFetchingPrices else { return }
-        isFetchingPrices = true
-
-        firstly {
-            provider.request(priceToUpdate)
-        }.map { response -> [AlphaWallet.Address: CoinTicker] in
-            let tickers = try response.map([CoinTicker].self, using: JSONDecoder())
-            let tempTickers = tickers.reduce([String: CoinTicker]()) { (dict, ticker) -> [String: CoinTicker] in
-                var dict = dict
-                dict[ticker.contract] = ticker
-                return dict
-            }
-            var resultTickers = [AlphaWallet.Address: CoinTicker]()
-            for (contract, ticker) in tempTickers {
-                guard let contractAddress = AlphaWallet.Address(uncheckedAgainstNullAddress: contract) else { continue }
-                resultTickers[contractAddress] = ticker
-            }
+        let objects = enabledObject.map { TokenMappedToTicker(tokenObject: $0) }
 
-            return resultTickers
-        }.done { [weak self] tickers in
-            self?.tickers = tickers
-            self?.updateDelegate()
-        }.cauterize().finally { [weak self] in
-            self?.isFetchingPrices = false
-        }
-    }
-
-    private func getPriceToUpdate() -> AlphaWalletService? {
-        switch server {
-        case .main:
-            return .priceOfEth(config: config)
-        case .xDai:
-            return .priceOfDai(config: config)
-        case .kovan, .ropsten, .rinkeby, .poa, .sokol, .classic, .callisto, .goerli, .artis_sigma1, .artis_tau1, .binance_smart_chain, .binance_smart_chain_testnet, .custom, .heco, .heco_testnet, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .polygon, .mumbai_testnet:
-            return nil
+        coinTickersResolver.coinTickers(enabledObject: objects).done { _ in
```

Comment:
`Promise<Void>` doesn't need an arg for the closure.

Reply:
Actually, if we don't intend to act on the promise, maybe `coinTickers(enabledObject:)` should just have no return value?